packages/devtools_app/test/flutter/extent_delegate_list_test.dart

```diff
@@ -0,0 +1,238 @@
+        (trailingScrollOffset - leadingScrollOffset) /
+        (lastIndex - firstIndex + 1);
+  }
+
+  @override
+  int get childCount => children.length;
+
+  @override
+  void didAdoptChild(RenderBox child) {
+    assert(_currentlyUpdatingChildIndex != null);
+    final SliverMultiBoxAdaptorParentData childParentData =
+        child.parentData as SliverMultiBoxAdaptorParentData;
+    childParentData.index = _currentlyUpdatingChildIndex;
+  }
+
+  @override
+  void setDidUnderflow(bool value) {}
+}
+
+void main() {
+  group('RenderSliverFixedExtentList', () {
+    group('extentDelegate', () {
+      test('itemExtent', () {
+        final extents = [100.0, 200.0, 50.0, 100.0];
+        final extentDelegate = FixedExtentDelegate(
+            // Create items with increasing extents.
+            computeExtent: (index) => extents[index],
+            computeLength: () => extents.length);
+
+        expect(extentDelegate.length, equals(4));
+        for (int i = 0; i < extents.length; i++) {
+          expect(extentDelegate.itemExtent(i), extents[i]);
+        }
+        expect(extentDelegate.itemExtent(1), 200.0);
+        extents[1] = 500.0;
+        extentDelegate.recompute();
+        expect(extentDelegate.itemExtent(1), 500.0);
+      });
+
+      test('getMinChildIndexForScrollOffset', () {
+        final extents = [100.0, 200.0, 50.0, 100.0];
+        final extentDelegate = FixedExtentDelegate(
+            // Create items with increasing extents.
+            computeExtent: (index) => extents[index],
+            computeLength: () => extents.length);
+
+        expect(extentDelegate.minChildIndexForScrollOffset(0), 0);
+        expect(extentDelegate.minChildIndexForScrollOffset(-1000), 0);
+        expect(extentDelegate.minChildIndexForScrollOffset(99), 0);
+        expect(extentDelegate.minChildIndexForScrollOffset(99.99999999999), 1);
+        expect(extentDelegate.minChildIndexForScrollOffset(250), 1);
+        expect(extentDelegate.minChildIndexForScrollOffset(299), 1);
+        expect(extentDelegate.minChildIndexForScrollOffset(299.99999999999), 2);
+        expect(extentDelegate.minChildIndexForScrollOffset(300), 2);
+        expect(extentDelegate.minChildIndexForScrollOffset(330), 2);
+        expect(extentDelegate.minChildIndexForScrollOffset(350), 3);
+        expect(extentDelegate.minChildIndexForScrollOffset(449), 3);
+        // Off the end of the list.
+        expect(extentDelegate.minChildIndexForScrollOffset(450), 4);
+        expect(extentDelegate.minChildIndexForScrollOffset(1000000), 4);
+      });
+
+      try {
+        test('getMaxChildIndexForScrollOffset', () {
+          final extents = [100.0, 200.0, 50.0, 100.0];
+          final extentDelegate = FixedExtentDelegate(
+              // Create items with increasing extents.
+              computeExtent: (index) => extents[index],
+              computeLength: () => extents.length);
+
+          // The behavior is a bit counter intuitive but this matching the
```

Comment:
nit: move this comment to the case it applies to (looks like it is the 4th expect statement and the second to last?)

Reply:
done.